sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventService.java

```diff
@@ -145,6 +145,44 @@ public EventService() {
 		return em.createQuery(cq).getResultList();
 	}
 
+	public List<Event> getAllByCaseAfterDateForNotification(Case caze, Date date) {
+		if (caze == null || date == null) {
+			return Collections.emptyList();
+		}
+
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<Event> cq = cb.createQuery(Event.class);
+		Root<EventParticipant> from = cq.from(EventParticipant.class);
+		Join<EventParticipant, Event> eventJoin = from.join(EventParticipant.EVENT, JoinType.INNER);
+
+		Predicate diseaseFilter = cb.equal(eventJoin.get(Event.DISEASE), caze.getDisease());
+		Predicate personFilter = cb.equal(from.get(EventParticipant.PERSON), caze.getPerson());
+
+		Timestamp timestamp = DateHelper.toTimestampUpper(date);
+		Predicate dateFilter = cb.or(
+				CriteriaBuilderHelper.greaterThanAndNotNull(cb, eventJoin.get(Event.START_DATE), timestamp),
+				cb.and(
+						cb.isNotNull(eventJoin.get(Event.END_DATE)),
+						CriteriaBuilderHelper.greaterThanAndNotNull(cb, eventJoin.get(Event.END_DATE), timestamp)));
+
+		Predicate surveillanceOfficerFilter = cb.and(
+				cb.isNotNull(eventJoin.get(Event.SURVEILLANCE_OFFICER)),
+				cb.not(cb.equal(eventJoin.get(Event.SURVEILLANCE_OFFICER), caze.getReportingUser())));
+
+		cq.where(
+				cb.and(
+						diseaseFilter,
+						personFilter,
+						dateFilter,
+						surveillanceOfficerFilter,
+						cb.isFalse(eventJoin.get(Event.DELETED)),
```

Comment:
Can you use `createActiveEventsFilter` here or isn't that possible because you're using event participant as root?